src/test/java/com/artipie/gem/GemITCase.java

```diff
@@ -121,18 +131,25 @@ void tearDown() {
     }
 
     private void push(final boolean anonymous) throws Exception {
-        final String key = new Base64Encoded("any:any").asString();
         new TestResource(String.format("gem/%s", GemITCase.RAILS))
             .saveTo(this.storage, new Key.From(GemITCase.RAILS));
         final String tmpurl = this.url.string(anonymous);
-        this.cntn.execStdout(
-            "/bin/bash", "-c",
-            String.format(
-                "GEM_HOST_API_KEY=%s gem push %s --host %s",
-                key, GemITCase.RAILS,
-                tmpurl.substring(0, tmpurl.length() - 1)
-            )
-        );
+        if (anonymous) {
+            final String key = new Base64Encoded("any:any").asString();
+            this.cntn.execStdout(
+                "/bin/bash", "-c",
+                String.format(
+                    "GEM_HOST_API_KEY=%s gem push %s --host %s",
+                    key, GemITCase.RAILS,
+                    tmpurl.substring(0, tmpurl.length() - 1)
+                )
+            );
+        } else {
+            this.cntn.execStdout(
+                "gem", "push", GemITCase.RAILS,
+                "--host", tmpurl.substring(0, tmpurl.length() - 1)
+            );
+        }
```

Comment:
@genryxy it seems odd that there is no `GEM_HOST_API_KEY` specified in case of non-anonymous operation

Reply:
@olegmoz  I think that credentials in this case should be obtained from the url that looks like `http://user:pass@host:port/my-gem`

Reply:
@genryxy could you please prove it by some link to tutorial or docs? Or maybe there is existing test in gem-adapter repository? I am under impression that `GEM_HOST_API_KEY` is the only authentication option supported by `gem` CLI tool